snapd: updates

bboozzoo opened the pull request:
@aimileus thanks for updating the package to 2.31!

@zyga @aimileus some fixes and updates:

-  add missing build dependency on libseccomp

- add bash-completion as optional dependency

- drop python-docutils from makedepends (we don't use them, not sure why they
  were left behind)

- update mkversion to include $pkgrel, this affects what is reported in `snap
  version`:
```
    snap    2.31-2
    snapd   2.31-2
    series  16
    arch
    kernel  4.15.1-2-ARCH
```
- dump extra cleanups in pre_remove(), those are already done by snap-mgmt

- pass all configuration settings when running `make -C data`

bboozzoo committed:
snapd: updates

Some updates:

- add missing dependency on libseccomp

- add bash-completion as optional dependency

- drop python-docutils from makedepends (we don't use them, not sure why they
  were left behind)

- update mkversion to include $pkgrel, this affects what is reported in `snap
  version`:
    snap    2.31-2
    snapd   2.31-2
    series  16
    arch
    kernel  4.15.1-2-ARCH

- dump extra cleanups in pre_remove(), those are already done by snap-mgmt

- pass all configuration settings when running `make -C data`

Signed-off-by: Maciek Borzecki <maciek.borzecki@gmail.com>

bboozzoo commented:
Let me do a quick amend, there's some code in snapd.install that can go away now.

zyga commented on .SRCINFO:
```diff
@@ -1,21 +1,22 @@
 	makedepends = go-tools
+	makedepends = libseccomp
 	makedepends = libcap
-	makedepends = python-docutils
```
This was used for creating manual pages for snap-confine.

zyga commented on .SRCINFO:
```diff
@@ -1,21 +1,22 @@
 	depends = squashfs-tools
 	depends = libseccomp
 	depends = libsystemd
+	optdepends = bash-completion: bash completion support
```
Not a strong feeling but does it have to be optional? Is it such a big dependency to avoid?

zyga reviewed (approved):
LGTM with an open question on the python build dependency.

bboozzoo committed:
snapd: fix build with go1.9.4

Drop all cgo/pkgconfig flags in snap-seccomp that are now rejected by go1.9.4+.

Signed-off-by: Maciek Borzecki <maciek.borzecki@gmail.com>

bboozzoo commented:
hope you don't mind guys, i'll be pushing this to AUR

bboozzoo commented:
Pushed as https://aur.archlinux.org/cgit/aur.git/commit/?h=snapd&id=57cf6441a75d1d01e0c720bdb77f875e87bd4dc7
